torch/autograd/grad_mode.py

```diff
@@ -54,7 +54,39 @@ def __init__(self):
         self.prev = torch.is_grad_enabled()
 
     def __enter__(self):
-        torch.set_grad_enabled(True)
+        torch._C.set_grad_enabled(True)
+
+    def __exit__(self, *args):
+        torch.set_grad_enabled(self.prev)
+        return False
+
+
+class set_grad_enabled(object):
```

Comment:
Can you please add this function to the docs (next to `enable_grad`) and add a simple test for both ways this can be used?

Reply:
I created a section in the autograd docs with no_grad, enable_grad and set_grad_enabled.